### PR TITLE
vSphere Refactoring code - Connection object

### PR DESCRIPTION
**Pull request by divyenpatel**
This PR is part of refactoring changes for Connection object.
It includes function to connect to the vCenter and to get Datacenter object.

@luomiao @rohitjogvmw @tusharnt @BaluDontu

### CODE REVIEW

File: pkg/cloudprovider/providers/vsphere/vclib/connection.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	if err != nil {
+		glog.Errorf("Failed to create govmomi client. err: %+v", err)
+		return err
+	}
+	return nil
+}
+
+// GetDatacenter returns the DataCenter Object for the given datacenterPath
+// If datacenter is located in a folder, include full path to datacenter else just provide the datacenter name
+func (connection *VSphereConnection) GetDatacenter(ctx context.Context, datacenterPath string) (*Datacenter, error) {
```
Comment (rohitjogvmw) [inline pkg/cloudprovider/providers/vsphere/vclib/connection.go]: Hi Divyen,

Should this function belong to Datacenter file?

Comment (divyenpatel) [inline pkg/cloudprovider/providers/vsphere/vclib/connection.go]: if we keep this in the Datacenter Object, then we will need to pass connection *VSphereConnection, as an argument to GetDatacenter.

I am fine with either.

Comment (rohitjogvmw) [inline pkg/cloudprovider/providers/vsphere/vclib/connection.go]: I can imagine the complexity by adding this method to Datacenter.
For now it is ok to keep it here.

**Comment by luomiao** [comment]
LGTM

**Pull request merged**